gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/DistributedRegion.java

```diff
@@ -1620,6 +1620,7 @@ protected void cleanUpAfterFailedGII(boolean recoverFromDisk) {
     } finally {
       try {
         doPostEntryDestructionCleanup(true, null, false, giiIndexLockAcquired);
+        writeUnlockEnqueueDelta();
```

Comment:
Where is the corresponding lock call? Are we sure that this will be always locked here? Also this should go into the finally block below.